From 45137eff5ceb46c6157dc3b6d5c670e1c267c540 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 15 Sep 2026 22:54:29 +0200
Subject: [PATCH] avformat/asfdec_f: drop the stream count limit

Found during triage/review of the security report M3Ib1Zm6OX8V
---
 libavformat/asfdec_f.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/libavformat/asfdec_f.c b/libavformat/asfdec_f.c
index 212ad449649e..d02c843ba626 100644
--- a/libavformat/asfdec_f.c
+++ b/libavformat/asfdec_f.c
@@ -134,7 +134,6 @@ static const AVClass asf_class = {
 #undef NDEBUG
 #include <assert.h>
 
-#define ASF_MAX_STREAMS 127
 #define FRAME_HEADER_SIZE 6
 // Fix Me! FRAME_HEADER_SIZE may be different.
 // (7 is known to be too large for GipsyGuitar.wmv)
@@ -311,11 +310,6 @@ static int asf_read_stream_properties(AVFormatContext *s, int64_t size)
     int64_t pos1, pos2, start_time;
     int test_for_ext_stream_audio, is_dvr_ms_audio = 0;
 
-    if (s->nb_streams == ASF_MAX_STREAMS) {
-        av_log(s, AV_LOG_ERROR, "too many streams\n");
-        return AVERROR(EINVAL);
-    }
-
     pos1 = avio_tell(pb);
 
     st = avformat_new_stream(s, NULL);
